pkg/model/manifest.go

```diff
@@ -722,6 +722,8 @@ func (m *Manifest) setDefaults() error {
 			b.Context = b.Name
 			b.Name = ""
 		}
+		b.setDockerfileDefaults()
+
 		if !(b.Image != "" && len(b.VolumesToInclude) > 0 && b.Dockerfile == "") {
 			b.setBuildDefaults()
```

Comment:
I think it may be easier to combine all the b.*Defaults into a function tis is simply always called at this point and all instantiation logic is placed there. As it stands it looks like we set things up in 3 different places here. 

It may also be better to do this via a New* whenever it's created with the `m` or `manifest`. It may be that a New* isn't viable if we are relying on the zero value initialization.

Reply:
I think we should simplify the default setting and add test coverage for all of it.

Reply:
I will try it. Problem is that the setBuildDefaults function is beeing called from a lot of places... That's why I didn't want to touch anything